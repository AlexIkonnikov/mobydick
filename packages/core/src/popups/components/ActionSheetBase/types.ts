import {StyleProp, ViewStyle} from 'react-native';
import React, {PropsWithChildren, ReactElement} from 'react';

import {TypographyProp} from '../../../typography/types';
import {IPopup} from '../../types';

export interface IPropsContents {
  title: string;
  textFont?: TypographyProp;
  leftIcon?: ReactElement | undefined;
  onPress?(): void;
  radio?: string;
  checkboxList?: string[];
  disabled?: boolean;
  itemType: IItemType;
}

export enum IItemType {
  firstItem = 'firstItem',
  innerItem = 'innerItem',
  lastItem = 'lastItem',
  singleItem = 'singleItem',
  cancelItem = 'cancelItem',
}

export interface IPropsItem extends IPropsContents {
  style?: ViewStyle;
  isStatusPressedForTest?: boolean;
}

export interface IActionSheetBaseProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  overlayStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  children: React.ReactNode;
}
