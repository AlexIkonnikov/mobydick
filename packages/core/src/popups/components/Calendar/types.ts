export interface IChangeDate {
  dateStart: string;
  dateEnd: string;
}

export enum IButtonView {
  small = 'small',
  large = 'large',
}
