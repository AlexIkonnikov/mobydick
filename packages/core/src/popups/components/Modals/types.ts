import {IButtonTypes} from '../../../cta';

export interface IModalProps {
  title: string;
  descriptionText: string;
  buttonText?: string;
}

export interface IModalAskProps {
  title: string;
  descriptionText: string;

  typeLeft?: IButtonTypes;
  textLeft?: string;

  onPressRight(): void;
  typeRight?: IButtonTypes;
  textRight?: string;
}
