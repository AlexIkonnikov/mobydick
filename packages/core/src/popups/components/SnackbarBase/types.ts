import {StyleProp, ViewStyle} from 'react-native';
import {PropsWithChildren} from 'react';

import {IPopup, IPosition} from '../../types';

export interface ISnackbarProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  containerStyle?: StyleProp<ViewStyle>;
  position: IPosition;
  timeShow?: number;
}
