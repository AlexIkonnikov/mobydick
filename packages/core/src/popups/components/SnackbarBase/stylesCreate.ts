import {StyleSheet} from 'react-native';

import {IThemeContext} from '../../../styles/theme/types';
import {IPosition} from '../../types';

const stylesCreate = (theme: IThemeContext, placement?: IPosition) => {
  const {colors} = theme;

  const placementStyle = () => {
    switch (placement) {
      case IPosition.top:
        return {
          top: 20,
        };
      case IPosition.bottom:
      default:
        return {
          bottom: 10,
        };
    }
  };
  return StyleSheet.create({
    overlayStyle: {
      backgroundColor: 'transparent',
      position: 'absolute',
      width: '100%',
      flex: 1,
      alignItems: 'center',
      ...placementStyle(),
    },
    container: {
      backgroundColor: colors.BgContrast,
      borderRadius: theme.spaces.Space12,
      justifyContent: 'center',
      alignItems: 'center',
      flexDirection: 'row',
      paddingHorizontal: theme.spaces.Space20,
      paddingVertical: theme.spaces.Space16,
      margin: theme.spaces.Space20,
    },
    title: {
      flex: 1,
      paddingRight: theme.spaces.Space8,
    },
  });
};

export default stylesCreate;
