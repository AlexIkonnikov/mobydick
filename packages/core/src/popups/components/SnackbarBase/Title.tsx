import {FC} from 'react';

import useStyles from '../../../styles/hooks/useStyles';
import {Typography} from '../../../typography/components/Typography/Typography';
import {ITitlePopup} from '../PopupBase';

import stylesCreate from './stylesCreate';

const Title: FC<ITitlePopup> = props => {
  const [styles] = useStyles(stylesCreate);
  const {title, titleStyles, titleFont} = props;

  return (
    <Typography
      style={[styles.title, titleStyles]}
      font={titleFont ? titleFont : 'Medium-Contrast-XS'}>
      {title}
    </Typography>
  );
};

export default Title;
