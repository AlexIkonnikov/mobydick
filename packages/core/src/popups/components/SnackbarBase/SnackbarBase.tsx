import {FC, useEffect, useRef} from 'react';

import {returnTrue} from '../../functions';
import useStyles from '../../../styles/hooks/useStyles';
import View from '../../../basic/components/View/View';

import Title from './Title';
import stylesCreate from './stylesCreate';
import {ISnackbarProps} from './types';

const DEFAULT_TIME_SHOW = 5000;

const SnackbarBase: FC<ISnackbarProps> & {Title: typeof Title} = props => {
  const {children, onClose, containerStyle, overlayStyle, position, timeShow} =
    props;
  const [styles] = useStyles(stylesCreate, position);

  const timeout = useRef<ReturnType<typeof setTimeout> | null>(null);

  useEffect(() => {
    timeout.current = setTimeout(
      () => {
        onClose();
      },
      timeShow ? timeShow : DEFAULT_TIME_SHOW,
    );

    return () => {
      timeout.current && clearTimeout(timeout.current);
    };
  }, []);

  return (
    <View style={[styles.overlayStyle, overlayStyle]}>
      <View
        style={[styles.container, containerStyle]}
        onStartShouldSetResponder={returnTrue}>
        {children}
      </View>
    </View>
  );
};

SnackbarBase.Title = Title;
export default SnackbarBase;
