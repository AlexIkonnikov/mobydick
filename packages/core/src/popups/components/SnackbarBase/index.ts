import SnackbarBase from './SnackbarBase';

export {SnackbarBase};
export * from './types';
