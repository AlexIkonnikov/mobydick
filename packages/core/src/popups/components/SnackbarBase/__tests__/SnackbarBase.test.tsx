import React from 'react';
import {render} from '@testing-library/react-native';

import SnackbarBase from '../SnackbarBase';
import {IPosition} from '../../../types';

describe('@lad-tech/mobydick-core/SnackbarBase', () => {
  jest.useFakeTimers();
  it('should renders correctly', () => {
    const onClose = jest.fn();
    const {toJSON} = render(
      <SnackbarBase id={'id'} onClose={onClose} position={IPosition.top} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly with timeShow', () => {
    const onClose = jest.fn();
    const {toJSON} = render(
      <SnackbarBase
        id={'id'}
        onClose={onClose}
        position={IPosition.bottom}
        timeShow={1000}
      />,
    );
    jest.runAllTimers();
    expect(toJSON()).toMatchSnapshot();
  });
});
