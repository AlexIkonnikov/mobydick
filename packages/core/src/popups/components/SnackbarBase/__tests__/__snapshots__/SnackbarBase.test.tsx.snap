// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-core/SnackbarBase should renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "position": "absolute",
        "top": 20,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#20242D",
          "borderRadius": 36,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 60,
          "paddingHorizontal": 60,
          "paddingVertical": 48,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`@npm/mobydick-core/SnackbarBase should renders correctly with timeShow 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "bottom": 10,
        "flex": 1,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#20242D",
          "borderRadius": 36,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 60,
          "paddingHorizontal": 60,
          "paddingVertical": 48,
        },
        undefined,
      ]
    }
  />
</View>
`;
